dds/DCPS/QOS_XML_Handler/XML_MemBuf_Intf.cpp

```diff
@@ -0,0 +1,256 @@
+      parser_->setValidationConstraintFatal (true);
+
+      // Set resolver using auxiliary XML_Schema_Resolver
+      parser_->setEntityResolver(res_);
+
+      // Set XML Error handler
+      parser_->setErrorHandler(eh_);
+
+      // Parsing buffer
+      try {
+        ACE::debug(true);
+        parser_->parse(xmlBuf);
+      }
+      catch (const SAXParseException& toCatch) {
+        char* message = XMLString::transcode(toCatch.getMessage());
+        ACE_ERROR ((LM_ERROR,
+          ACE_TEXT ("QOS_XML_MemBuf_Handler::init - ")
+          ACE_TEXT ("Exception message is: <%C>\n"),
+          message));
+        XMLString::release(&message);
+        ACE::debug(false);
+        return DDS::RETCODE_ERROR;
+      }
+      catch (const XMLException& toCatch) {
+        char* message = XMLString::transcode(toCatch.getMessage());
+        ACE_ERROR ((LM_ERROR,
+          ACE_TEXT ("QOS_XML_MemBuf_Handler::init - ")
+          ACE_TEXT ("Exception message is: <%C>\n"),
+          message));
+        XMLString::release(&message);
+        ACE::debug(false);
+        return DDS::RETCODE_ERROR;
+      }
+      catch (const DOMException& toCatch) {
+        char* message = XMLString::transcode(toCatch.getMessage());
+        ACE_ERROR ((LM_ERROR,
+          ACE_TEXT ("QOS_XML_MemBuf_Handler::init - ")
+          ACE_TEXT ("Exception message is: <%C>\n"),
+          message));
+        XMLString::release(&message);
+        ACE::debug(false);
+        return DDS::RETCODE_ERROR;
+      }
+      catch (...) {
+        ACE_ERROR ((LM_ERROR,
+          ACE_TEXT ("QOS_XML_MemBuf_Handler::init - ")
+          ACE_TEXT ("Unexpected exception\n")
+          ));
+        ACE::debug(false);
+        return DDS::RETCODE_ERROR;
+      }
+      ACE::debug(false);
+
+      DOMDocument * initialDoc = parser_->getDocument();
+      if (initialDoc == 0)
+      {
+        if (DCPS_debug_level > 1)
+        {
+          ACE_ERROR ((LM_ERROR,
+            ACE_TEXT ("QOS_XML_MemBuf_Handler::init - ")
+            ACE_TEXT ("Failed to parse string\n")
+            ));
+        }
+        return DDS::RETCODE_ERROR;
+      }
+
+      // Find "dds" node using the opendds namespace
+      DOMNodeList * ddsNodeList = initialDoc->getElementsByTagNameNS(XMLString::transcode("http://www.omg.org/dds"),XMLString::transcode("dds"));
+
+      // Check if it could find tag and namespace
+      if (ddsNodeList->getLength() == 0)
+      {
+        if (DCPS_debug_level > 1)
+          {
+            ACE_ERROR ((LM_ERROR,
+              ACE_TEXT ("QOS_XML_MemBuf_Handler::init - ")
+              ACE_TEXT ("Could not find tag in namespace\n")
+              ));
+          }
+        return DDS::RETCODE_ERROR;
+      }
+
+      // Get the first node. It is expected to have only one
+      DOMNode * ddsNode = ddsNodeList->item(0);
+      if (DCPS_debug_level > 1)
+      {
+        char* message = XMLString::transcode(ddsNode->getNodeName());
+        ACE_ERROR ((LM_INFO,
```

Comment:
Should be ACE_DEBUG for a debug message

Reply:
Hi
I'll do it, but in QOS_XML_Loader.cpp it uses as I did.
Check following code:

  DDS::ReturnCode_t
  QOS_XML_Loader::init(const ACE_TCHAR* qos_profile)
  {
    if (!qos_profile)
      {
        if (DCPS_debug_level > 5)
          {
            ACE_ERROR((LM_ERROR,
              ACE_TEXT("QOS_XML_Loader::init - ")
              ACE_TEXT("Passed an empty qos_profile, returning.\n")));
          }

Cheers

Reply:
ACE_ERROR is for logging error messages, ACE_DEBUG for debug/info messages, both could be independently enabled/disabled at compilation time

Reply:
Another question is:
How can i set the DCPS_debug_level variable? I set grater than 1, but is there some policy to do that?
Cheers

Reply:
> Another question is:
> How can i set the DCPS_debug_level variable? I set grater than 1, but is there some policy to do that?
> Cheers

See http://download.objectcomputing.com/OpenDDS/OpenDDS-latest.pdf, chapter 7.6.1